sdk/Lusid.Sdk.Tests/tutorials/Properties/TransactionProperties.cs

```diff
@@ -0,0 +1,117 @@
+            string perpetualPropertyPropVal = "TestPropertyLabelValue";
+
+            EnsurePropertyDefinition(code);
+
+            var transaction = new TransactionRequest(
+                // unique transaction id
+                transactionId: Guid.NewGuid().ToString(),
+
+                // instruments must already exist in LUSID and have a valid LUSID instrument id
+                instrumentIdentifiers: new Dictionary<string, string>
+                {
+                    [TestDataUtilities.LusidInstrumentIdentifier] = _instrumentIds[0]
+                },
+
+                type: "Buy",
+                totalConsideration: new CurrencyAndAmount(1230, "GBP"),
+                transactionDate: effectiveDate,
+                settlementDate: effectiveDate,
+                units: 100,
+                transactionPrice: new TransactionPrice(12.3M),
+                source: "Custodian",
+
+                //Specification of the transaction property we are testing
+                properties: new Dictionary<string, PerpetualProperty>
+                            {
+                                { 
+                                    perpetualPropertyKey,
+                                    new PerpetualProperty(perpetualPropertyKey, new PropertyValue(labelValue: perpetualPropertyPropVal))
+                                }
+                            }
+            );
+
+            // add the transaction
+            UpsertPortfolioTransactionsResponse upsertResp =  _transactionPortfoliosApi.UpsertTransactions(TestDataUtilities.TutorialScope, _portfolioCode, new List<TransactionRequest> { transaction });
+
+            // get the transaction and verify the reponse
+            var transactions = _transactionPortfoliosApi.GetTransactions(TestDataUtilities.TutorialScope, _portfolioCode);
+
+            Assert.That(transactions.Values, Has.Count.EqualTo(1));
+            Assert.That(transactions.Values[0].TransactionId, Is.EqualTo(transaction.TransactionId));
+
+            // assert that the custom properties was upserted with the transaction and is returned
+            Assert.IsTrue(transactions.Values[0].Properties.ContainsKey(perpetualPropertyKey));
+            Assert.That(transactions.Values[0].Properties[perpetualPropertyKey].Value.LabelValue == perpetualPropertyPropVal);
+        }
+
+        private void EnsurePropertyDefinition(string code)
+        {
+            var propertyApi = _apiFactory.Api<IPropertyDefinitionsApi>();
+
+            try
+            {
+                propertyApi.GetPropertyDefinition("Transaction", TestDataUtilities.TutorialScope, code);
+            }
+            catch (ApiException)
+            {
+                //    Property definition doesn't exist (returns 404), so create one
+                //    Details of the property to be created
+                var propertyDefinition = new CreatePropertyDefinitionRequest(
```

Comment:
i would check for the 404 before creating

Reply:
Done. This code was originally copied from Instruments.cs and modified slightly so that the domain was Transaction rather than Instrument. 

I've just checked in a version that explicitly checks for errorcode == 404. I now also pass the domain as a parameter as I though we might extract this to TestDataUtilities.cs thereby also allowing the code to be removed from Instruments.cs. I haven't done this yet as I didn't know how you feel about having tutorial code in the Lusid.Sdk.Tests.Utilities namespace.